src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -94,6 +94,13 @@
      */
     @Getter
     public boolean backpointersDisabled = false;
+    /**
+     * To prevent livelock condition in fetchLayout.
+     * Notifies that the runtime is no longer used
+     * and retries to fetch the layout can be stopped.
```

Comment:
Timeout will not be sufficient, there can be several layout servers in the list to which the runtime connects to. So there is no definite timeout value we can put there.
We can have retries instead. Lets say 3 complete rounds of all the layout servers.
This will delay the shutdown but I agree it will be a cleaner way to handle this.